packages/nodes-base/nodes/Aws/DynamoDB/AwsDynamoDB.node.ts

```diff
@@ -0,0 +1,329 @@
+				} else {
+					body.Key = populatePartitionKey.call(this, i);
+				}
+
+				const { readConsistencyModel } = this.getNodeParameter('additionalFields', i) as {
+					readConsistencyModel: 'eventuallyConsistent' | 'stronglyConsistent';
+				};
+
+				if (readConsistencyModel) {
+					body.ConsistentRead = readConsistencyModel === 'stronglyConsistent';
+				}
+
+				const headers = {
+					'Content-Type': 'application/x-amz-json-1.0',
+					'X-Amz-Target': 'DynamoDB_20120810.GetItem',
+				};
+
+				responseData = await awsApiRequestREST.call(this, 'dynamodb', 'POST', '/', JSON.stringify(body), headers);
+
+				if (!responseData.Item) {
+					responseData = [];
+				}
+
+				const simple = this.getNodeParameter('simple', 0) as boolean;
+
+				if (simple) {
+					responseData = simplify(responseData.Item);
+				}
+
+			} else if (operation === 'query') {
+
+				// ----------------------------------
+				//             query
+				// ----------------------------------
+
+				// https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html
+
+				const eavUi = this.getNodeParameter('expressionAttributeValues.details', i) as IAttributeValueUi[];
+
+				const body: IRequestBody = {
+					TableName: this.getNodeParameter('tableName', i) as string,
+					KeyConditionExpression: this.getNodeParameter('keyConditionExpression', i) as string,
+					ExpressionAttributeValues: adjustExpressionAttributeValues(eavUi),
+				};
+
+				const {
+					indexName,
+					projectionExpression,
+					readConsistencyModel,
+				} = this.getNodeParameter('additionalFields', i) as {
+					indexName: string;
+					projectionExpression: string;
+					readConsistencyModel: 'eventuallyConsistent' | 'stronglyConsistent';
+				};
+
+				if (indexName) {
+					body.IndexName = indexName;
+				}
+
+				if (projectionExpression) {
+					body.ProjectionExpression = projectionExpression;
+				}
+
+				if (readConsistencyModel) {
+					body.ConsistentRead = readConsistencyModel === 'stronglyConsistent';
+				}
+
+				const headers = {
+					'Content-Type': 'application/x-amz-json-1.0',
+					'X-Amz-Target': 'DynamoDB_20120810.Query',
+				};
+
+				responseData = await awsApiRequestREST.call(this, 'dynamodb', 'POST', '/', JSON.stringify(body), headers);
+
+				if (responseData.Items) {
+					responseData = responseData.Items.map(simplify);
+				}
+
+			} else if (operation === 'scan') {
+
+				// ----------------------------------
+				//             scan
+				// ----------------------------------
+
+				// https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Scan.html
+
+				const eavUi = this.getNodeParameter('expressionAttributeValues.details', i) as IAttributeValueUi[];
+
+				const body: IRequestBody = {
+					TableName: this.getNodeParameter('tableName', i) as string,
+					FilterExpression: this.getNodeParameter('filterExpression', i) as string,
+					ExpressionAttributeValues: adjustExpressionAttributeValues(eavUi),
```

Comment:
ExpressionAttributeValues on "scan" operation is optional. 

Just wanted to add a bugfix to my branch, but you somehow already refactored it. :sweat_smile: